src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -21,11 +21,23 @@ namespace ILCompiler
 {
     public class CompilationOptions
     {
-        public IReadOnlyDictionary<string, string> InputFilePaths;
-        public IReadOnlyDictionary<string, string> ReferenceFilePaths;
+        /// <summary>
+        /// Associate for each simple name, the list of assembly full path names matching the simple name.
+        /// Useful when including multiple assemblies with same name from different locations.
```

Comment:
This is correct. I had to comment out the throwing of an exception to allow this pattern and put a todo there:
https://github.com/manu-silicon/corert/commit/d0802488b081813802e543b2911d8428b3f1c084#diff-ae678a50ed062f0d32887887b6333028R215